desc/geometry/curve.py

```diff
@@ -817,8 +827,21 @@ def __init__(
             errorif(knots[-1] > 2 * np.pi, ValueError, "knots must lie in [0, 2pi]")
             knots = knots[:-1] if closed_flag else knots
 
+        if break_indices is None:
+            unbroken_spline_intervals = [[]]
+        else:
+            # check that input is monotonic
+            assert np.all(
+                break_indices == np.unique(sorted(break_indices))
+            ), "Indices must be monotonic."
+            unbroken_spline_intervals = [
+                [break_indices[i - 1], break_indices[i]]
+                for i in range(len(break_indices))
+            ]
+
         self._knots = knots
-        self.method = method
+        self._method = method
+        self.intervals = unbroken_spline_intervals
```

Comment:
Does this have to be a "public" attribute? (i.e., no leading underscore) Probably best to make it private and add a getter method like we do for everything else

Also why call it `unbroken_spline_intervals` and then rename the variable to `intervals`? Might be better to just use `intervals` the whole time, like we do for knots, method, etc.